src/frontend/src/icp-eth/components/tokens/AddTokenByNetwork.svelte

```diff
@@ -23,11 +24,13 @@
 	let indexCanisterId: string;
 	let erc20ContractAddress: string;
 
-	$: tokenData = {
-		ledgerCanisterId,
-		indexCanisterId,
-		erc20ContractAddress
-	};
+	onDestroy(() => {
+		tokenData = isNetworkIdICP(network?.id)
```

Comment:
Can you add a comment about why we do this as it's quite an original pattern?

Reply:
check now, I updated the logic, maybe it is better?